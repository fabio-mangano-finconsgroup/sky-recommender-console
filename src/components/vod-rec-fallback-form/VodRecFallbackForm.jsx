import React from 'react';
import { Formik, Form } from 'formik';
import { styled } from '@mui/material';
import Grid from '@mui/material/Grid';
import LoadingButton from '@mui/lab/LoadingButton';
import { DEFAULT_VALUES, validationSchema } from './validation';
import EventSlot from '../form/event-slot/EventSlot';
import SearchVodRec from '../../containers/search-vod-rec/SearchVodRec';
import Clear from '../form/clear/Clear';
import Modal from '../modal/Modal';

const VocRecFallbackFormWrapper = styled('div')({
  maxWidth: '100%',
  width: '700px',
  height: '100%',
  maxHeight: '600px',
  paddingTop: '10px',
});

const SlotsRowWrapper = styled('div')({
  display: 'flex',
  justifyContent: 'space-between',
});

const ButtonsWrapper = styled('div')({
  display: 'flex',
  justifyContent: 'space-between',
});

const VocRecFallbackForm = ({ onSubmit, initialValues, isSubmitting }) => {
  const [open, setOpen] = React.useState(false);
  const [currentSlot, setCurrentSlot] = React.useState(null);

  const handleOpen = (slotName) => {
    setCurrentSlot(slotName);
    setOpen(true);
  };

  const handleClose = () => setOpen(false);

  const mergedInitialValues = React.useMemo(
    () => ({
      ...DEFAULT_VALUES,
      ...initialValues,
    }),
    [initialValues],
  );

  return (
    <VocRecFallbackFormWrapper>
      <Formik
        onSubmit={onSubmit}
        initialValues={mergedInitialValues}
        validationSchema={validationSchema}
      >
        {({ setFieldValue }) => (
          <Form>
            <Grid container spacing={1.5}>
              <Grid item xs={12}>
                <SlotsRowWrapper>
                  <EventSlot name="recommendation.1" handleOpen={handleOpen} />
                  <EventSlot name="recommendation.2" handleOpen={handleOpen} />
                  <EventSlot name="recommendation.3" handleOpen={handleOpen} />
                  <EventSlot name="recommendation.4" handleOpen={handleOpen} />
                  <EventSlot name="recommendation.5" handleOpen={handleOpen} />
                </SlotsRowWrapper>
              </Grid>

              <Grid item xs={12}>
                <SlotsRowWrapper>
                  <EventSlot name="recommendation.6" handleOpen={handleOpen} />
                  <EventSlot name="recommendation.7" handleOpen={handleOpen} />
                  <EventSlot name="recommendation.8" handleOpen={handleOpen} />
                  <EventSlot name="recommendation.9" handleOpen={handleOpen} />
                  <EventSlot name="recommendation.10" handleOpen={handleOpen} />
                </SlotsRowWrapper>
              </Grid>

              <Grid item xs={12}>
                <ButtonsWrapper>
                  <Clear initialValues={DEFAULT_VALUES} />
                  <LoadingButton
                    type="submit"
                    variant="contained"
                    color="success"
                    loading={isSubmitting}
                  >
                    Update
                  </LoadingButton>
                </ButtonsWrapper>
              </Grid>
            </Grid>

            <Modal title="SEARCH VOD" open={open} handleClose={handleClose}>
              <SearchVodRec
                addEvent={(value) => {
                  setFieldValue(currentSlot, {
                    ...value,
                    position: currentSlot.split('.')[1],
                  });
                }}
                handleClose={handleClose}
              />
            </Modal>
          </Form>
        )}
      </Formik>
    </VocRecFallbackFormWrapper>
  );
};

export default VocRecFallbackForm;
