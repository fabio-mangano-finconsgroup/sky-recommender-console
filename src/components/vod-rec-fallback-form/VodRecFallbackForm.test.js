import React from 'react';
import { render, fireEvent, act, screen } from '@testing-library/react';
import VodRecFallbackForm from './VodRecFallbackForm';
import  fallbackRecMock  from '../../../fixtures/fallback-recc-mock.json';

describe('VodRecFallbackForm', () => {
  let props;
  let onSubitMock = jest.fn();


  it('should render', () => {
    console.log(fallbackRecMock[0])
    props = {
      onSubmit: onSubitMock,
      initialValues: fallbackRecMock[0],
      isSubmitting: false,
    };

    render(<VodRecFallbackForm {...props} />);
    expect(document.body.childNodes[0].children).toMatchSnapshot();
  });

   it('should not save if empty', async () => {
     props = {
       onSubmit: onSubitMock,
       initialValues: { id: '5', type: 'FALLBACK', recommendation: [] },
       isSubmitting: false,
     };

     render(<VodRecFallbackForm {...props} />);
     const updateButton = screen.getByText(/update/i);

     await act(async () => {
       await fireEvent.click(updateButton);
     });

   expect(onSubitMock).not.toHaveBeenCalled();
   });







});
