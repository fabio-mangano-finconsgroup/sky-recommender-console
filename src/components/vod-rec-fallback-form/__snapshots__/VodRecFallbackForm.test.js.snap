// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VodRecFallbackForm should render 1`] = `
HTMLCollection [
  <div
    class="css-b33b91"
  >
    <form
      action="#"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1.5 css-acctgf-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1em9ut6-MuiGrid-root"
        >
          <div
            class="css-gg4vpm"
          >
            <div
              class="css-lccpzd"
              data-test="61GU2"
              data-test-slot="fallback-slot"
            >
              <div
                class="MuiBox-root css-1c6wt7s"
              >
                <button
                  class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-1ci55s3-MuiButtonBase-root-MuiIconButton-root"
                  tabindex="0"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-colorError MuiSvgIcon-fontSizeSmall css-1efkf0c-MuiSvgIcon-root"
                    data-testid="ClearIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                    />
                  </svg>
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
                
                <div
                  class="MuiBox-root css-1agjfwn"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-lusxqg-MuiSvgIcon-root"
                    data-testid="ImageNotSupportedIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="m21.9 21.9-8.49-8.49-9.82-9.82L2.1 2.1.69 3.51 3 5.83V19c0 1.1.9 2 2 2h13.17l2.31 2.31 1.42-1.41zM5 18l3.5-4.5 2.5 3.01L12.17 15l3 3H5zm16 .17L5.83 3H19c1.1 0 2 .9 2 2v13.17z"
                    />
                  </svg>
                </div>
              </div>
              <p
                class="MuiTypography-root MuiTypography-body1 MuiTypography-noWrap css-zy8lap-MuiTypography-root"
                data-test="event-title"
              >
                <span>
                  <b>
                    La valle incantata
                  </b>
                </span>
              </p>
              <p
                class="MuiTypography-root MuiTypography-body1 MuiTypography-noWrap css-zy8lap-MuiTypography-root"
                data-test="event-startProgram"
              >
                24.12.2021 13:00
              </p>
              <p
                class="MuiTypography-root MuiTypography-body1 MuiTypography-noWrap css-zy8lap-MuiTypography-root"
                data-test="event-endProgram"
              >
                24.12.2022 16:00
              </p>
            </div>
            <div
              class="css-lccpzd"
              data-test="61GU3"
              data-test-slot="fallback-slot"
            >
              <div
                class="MuiBox-root css-1c6wt7s"
              >
                <button
                  class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-1ci55s3-MuiButtonBase-root-MuiIconButton-root"
                  tabindex="0"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-colorError MuiSvgIcon-fontSizeSmall css-1efkf0c-MuiSvgIcon-root"
                    data-testid="ClearIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                    />
                  </svg>
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
                
                <div
                  class="MuiBox-root css-1agjfwn"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-lusxqg-MuiSvgIcon-root"
                    data-testid="ImageNotSupportedIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="m21.9 21.9-8.49-8.49-9.82-9.82L2.1 2.1.69 3.51 3 5.83V19c0 1.1.9 2 2 2h13.17l2.31 2.31 1.42-1.41zM5 18l3.5-4.5 2.5 3.01L12.17 15l3 3H5zm16 .17L5.83 3H19c1.1 0 2 .9 2 2v13.17z"
                    />
                  </svg>
                </div>
              </div>
              <p
                class="MuiTypography-root MuiTypography-body1 MuiTypography-noWrap css-zy8lap-MuiTypography-root"
                data-test="event-title"
              >
                <span>
                  <b>
                    The Truman Show 2
                  </b>
                </span>
              </p>
              <p
                class="MuiTypography-root MuiTypography-body1 MuiTypography-noWrap css-zy8lap-MuiTypography-root"
                data-test="event-startProgram"
              >
                24.12.2021 13:00
              </p>
              <p
                class="MuiTypography-root MuiTypography-body1 MuiTypography-noWrap css-zy8lap-MuiTypography-root"
                data-test="event-endProgram"
              >
                24.12.2022 16:00
              </p>
            </div>
            <div
              class="css-lccpzd"
              data-test="61GU4"
              data-test-slot="fallback-slot"
            >
              <div
                class="MuiBox-root css-1c6wt7s"
              >
                <button
                  class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-1ci55s3-MuiButtonBase-root-MuiIconButton-root"
                  tabindex="0"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-colorError MuiSvgIcon-fontSizeSmall css-1efkf0c-MuiSvgIcon-root"
                    data-testid="ClearIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                    />
                  </svg>
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
                
                <div
                  class="MuiBox-root css-1agjfwn"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-lusxqg-MuiSvgIcon-root"
                    data-testid="ImageNotSupportedIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="m21.9 21.9-8.49-8.49-9.82-9.82L2.1 2.1.69 3.51 3 5.83V19c0 1.1.9 2 2 2h13.17l2.31 2.31 1.42-1.41zM5 18l3.5-4.5 2.5 3.01L12.17 15l3 3H5zm16 .17L5.83 3H19c1.1 0 2 .9 2 2v13.17z"
                    />
                  </svg>
                </div>
              </div>
              <p
                class="MuiTypography-root MuiTypography-body1 MuiTypography-noWrap css-zy8lap-MuiTypography-root"
                data-test="event-title"
              >
                <span>
                  <b>
                    I mercenari
                  </b>
                </span>
              </p>
              <p
                class="MuiTypography-root MuiTypography-body1 MuiTypography-noWrap css-zy8lap-MuiTypography-root"
                data-test="event-startProgram"
              >
                24.12.2021 13:00
              </p>
              <p
                class="MuiTypography-root MuiTypography-body1 MuiTypography-noWrap css-zy8lap-MuiTypography-root"
                data-test="event-endProgram"
              >
                24.12.2022 16:00
              </p>
            </div>
            <div
              class="css-lccpzd"
              data-test="61GU5"
              data-test-slot="fallback-slot"
            >
              <div
                class="MuiBox-root css-1c6wt7s"
              >
                <button
                  class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-1ci55s3-MuiButtonBase-root-MuiIconButton-root"
                  tabindex="0"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-colorError MuiSvgIcon-fontSizeSmall css-1efkf0c-MuiSvgIcon-root"
                    data-testid="ClearIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                    />
                  </svg>
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
                
                <div
                  class="MuiBox-root css-1agjfwn"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-lusxqg-MuiSvgIcon-root"
                    data-testid="ImageNotSupportedIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="m21.9 21.9-8.49-8.49-9.82-9.82L2.1 2.1.69 3.51 3 5.83V19c0 1.1.9 2 2 2h13.17l2.31 2.31 1.42-1.41zM5 18l3.5-4.5 2.5 3.01L12.17 15l3 3H5zm16 .17L5.83 3H19c1.1 0 2 .9 2 2v13.17z"
                    />
                  </svg>
                </div>
              </div>
              <p
                class="MuiTypography-root MuiTypography-body1 MuiTypography-noWrap css-zy8lap-MuiTypography-root"
                data-test="event-title"
              >
                <span>
                  <b>
                    Rogue one
                  </b>
                </span>
              </p>
              <p
                class="MuiTypography-root MuiTypography-body1 MuiTypography-noWrap css-zy8lap-MuiTypography-root"
                data-test="event-startProgram"
              >
                24.12.2021 13:00
              </p>
              <p
                class="MuiTypography-root MuiTypography-body1 MuiTypography-noWrap css-zy8lap-MuiTypography-root"
                data-test="event-endProgram"
              >
                24.12.2022 16:00
              </p>
            </div>
            <div
              class="css-lccpzd"
              data-test="61P1"
              data-test-slot="fallback-slot"
            >
              <div
                class="MuiBox-root css-1c6wt7s"
              >
                <button
                  class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-1ci55s3-MuiButtonBase-root-MuiIconButton-root"
                  tabindex="0"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-colorError MuiSvgIcon-fontSizeSmall css-1efkf0c-MuiSvgIcon-root"
                    data-testid="ClearIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                    />
                  </svg>
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
                
                <div
                  class="MuiBox-root css-1agjfwn"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-lusxqg-MuiSvgIcon-root"
                    data-testid="ImageNotSupportedIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="m21.9 21.9-8.49-8.49-9.82-9.82L2.1 2.1.69 3.51 3 5.83V19c0 1.1.9 2 2 2h13.17l2.31 2.31 1.42-1.41zM5 18l3.5-4.5 2.5 3.01L12.17 15l3 3H5zm16 .17L5.83 3H19c1.1 0 2 .9 2 2v13.17z"
                    />
                  </svg>
                </div>
              </div>
              <p
                class="MuiTypography-root MuiTypography-body1 MuiTypography-noWrap css-zy8lap-MuiTypography-root"
                data-test="event-title"
              >
                <span>
                  <b>
                    Doom
                  </b>
                </span>
              </p>
              <p
                class="MuiTypography-root MuiTypography-body1 MuiTypography-noWrap css-zy8lap-MuiTypography-root"
                data-test="event-startProgram"
              >
                24.12.2021 13:00
              </p>
              <p
                class="MuiTypography-root MuiTypography-body1 MuiTypography-noWrap css-zy8lap-MuiTypography-root"
                data-test="event-endProgram"
              >
                24.12.2022 16:00
              </p>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1em9ut6-MuiGrid-root"
        >
          <div
            class="css-gg4vpm"
          >
            <div
              class="css-lccpzd"
              data-test="61GP2"
              data-test-slot="fallback-slot"
            >
              <div
                class="MuiBox-root css-1c6wt7s"
              >
                <button
                  class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-1ci55s3-MuiButtonBase-root-MuiIconButton-root"
                  tabindex="0"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-colorError MuiSvgIcon-fontSizeSmall css-1efkf0c-MuiSvgIcon-root"
                    data-testid="ClearIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                    />
                  </svg>
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
                
                <div
                  class="MuiBox-root css-1agjfwn"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-lusxqg-MuiSvgIcon-root"
                    data-testid="ImageNotSupportedIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="m21.9 21.9-8.49-8.49-9.82-9.82L2.1 2.1.69 3.51 3 5.83V19c0 1.1.9 2 2 2h13.17l2.31 2.31 1.42-1.41zM5 18l3.5-4.5 2.5 3.01L12.17 15l3 3H5zm16 .17L5.83 3H19c1.1 0 2 .9 2 2v13.17z"
                    />
                  </svg>
                </div>
              </div>
              <p
                class="MuiTypography-root MuiTypography-body1 MuiTypography-noWrap css-zy8lap-MuiTypography-root"
                data-test="event-title"
              >
                <span>
                  <b>
                    La valle incantata 2
                  </b>
                </span>
              </p>
              <p
                class="MuiTypography-root MuiTypography-body1 MuiTypography-noWrap css-zy8lap-MuiTypography-root"
                data-test="event-startProgram"
              >
                24.12.2021 13:00
              </p>
              <p
                class="MuiTypography-root MuiTypography-body1 MuiTypography-noWrap css-zy8lap-MuiTypography-root"
                data-test="event-endProgram"
              >
                24.12.2022 16:00
              </p>
            </div>
            <div
              class="css-lccpzd"
              data-test="61GP3"
              data-test-slot="fallback-slot"
            >
              <div
                class="MuiBox-root css-1c6wt7s"
              >
                <button
                  class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-1ci55s3-MuiButtonBase-root-MuiIconButton-root"
                  tabindex="0"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-colorError MuiSvgIcon-fontSizeSmall css-1efkf0c-MuiSvgIcon-root"
                    data-testid="ClearIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                    />
                  </svg>
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
                
                <div
                  class="MuiBox-root css-1agjfwn"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-lusxqg-MuiSvgIcon-root"
                    data-testid="ImageNotSupportedIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="m21.9 21.9-8.49-8.49-9.82-9.82L2.1 2.1.69 3.51 3 5.83V19c0 1.1.9 2 2 2h13.17l2.31 2.31 1.42-1.41zM5 18l3.5-4.5 2.5 3.01L12.17 15l3 3H5zm16 .17L5.83 3H19c1.1 0 2 .9 2 2v13.17z"
                    />
                  </svg>
                </div>
              </div>
              <p
                class="MuiTypography-root MuiTypography-body1 MuiTypography-noWrap css-zy8lap-MuiTypography-root"
                data-test="event-title"
              >
                <span>
                  <b>
                    Se mi lasci ti cancello
                  </b>
                </span>
              </p>
              <p
                class="MuiTypography-root MuiTypography-body1 MuiTypography-noWrap css-zy8lap-MuiTypography-root"
                data-test="event-startProgram"
              >
                24.12.2021 13:00
              </p>
              <p
                class="MuiTypography-root MuiTypography-body1 MuiTypography-noWrap css-zy8lap-MuiTypography-root"
                data-test="event-endProgram"
              >
                24.12.2022 16:00
              </p>
            </div>
            <div
              class="css-lccpzd"
              data-test="61GP4"
              data-test-slot="fallback-slot"
            >
              <div
                class="MuiBox-root css-1c6wt7s"
              >
                <button
                  class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-1ci55s3-MuiButtonBase-root-MuiIconButton-root"
                  tabindex="0"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-colorError MuiSvgIcon-fontSizeSmall css-1efkf0c-MuiSvgIcon-root"
                    data-testid="ClearIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                    />
                  </svg>
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
                
                <div
                  class="MuiBox-root css-1agjfwn"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-lusxqg-MuiSvgIcon-root"
                    data-testid="ImageNotSupportedIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="m21.9 21.9-8.49-8.49-9.82-9.82L2.1 2.1.69 3.51 3 5.83V19c0 1.1.9 2 2 2h13.17l2.31 2.31 1.42-1.41zM5 18l3.5-4.5 2.5 3.01L12.17 15l3 3H5zm16 .17L5.83 3H19c1.1 0 2 .9 2 2v13.17z"
                    />
                  </svg>
                </div>
              </div>
              <p
                class="MuiTypography-root MuiTypography-body1 MuiTypography-noWrap css-zy8lap-MuiTypography-root"
                data-test="event-title"
              >
                <span>
                  <b>
                    I mercenari 2
                  </b>
                </span>
              </p>
              <p
                class="MuiTypography-root MuiTypography-body1 MuiTypography-noWrap css-zy8lap-MuiTypography-root"
                data-test="event-startProgram"
              >
                24.12.2021 13:00
              </p>
              <p
                class="MuiTypography-root MuiTypography-body1 MuiTypography-noWrap css-zy8lap-MuiTypography-root"
                data-test="event-endProgram"
              >
                24.12.2022 16:00
              </p>
            </div>
            <div
              class="css-lccpzd"
              data-test="61GP5"
              data-test-slot="fallback-slot"
            >
              <div
                class="MuiBox-root css-1c6wt7s"
              >
                <button
                  class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-1ci55s3-MuiButtonBase-root-MuiIconButton-root"
                  tabindex="0"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-colorError MuiSvgIcon-fontSizeSmall css-1efkf0c-MuiSvgIcon-root"
                    data-testid="ClearIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                    />
                  </svg>
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
                
                <div
                  class="MuiBox-root css-1agjfwn"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-lusxqg-MuiSvgIcon-root"
                    data-testid="ImageNotSupportedIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="m21.9 21.9-8.49-8.49-9.82-9.82L2.1 2.1.69 3.51 3 5.83V19c0 1.1.9 2 2 2h13.17l2.31 2.31 1.42-1.41zM5 18l3.5-4.5 2.5 3.01L12.17 15l3 3H5zm16 .17L5.83 3H19c1.1 0 2 .9 2 2v13.17z"
                    />
                  </svg>
                </div>
              </div>
              <p
                class="MuiTypography-root MuiTypography-body1 MuiTypography-noWrap css-zy8lap-MuiTypography-root"
                data-test="event-title"
              >
                <span>
                  <b>
                    VOD Scaduto
                  </b>
                </span>
              </p>
              <p
                class="MuiTypography-root MuiTypography-body1 MuiTypography-noWrap css-zy8lap-MuiTypography-root"
                data-test="event-startProgram"
              >
                24.12.2021 13:00
              </p>
              <p
                class="MuiTypography-root MuiTypography-body1 MuiTypography-noWrap css-zy8lap-MuiTypography-root"
                data-test="event-endProgram"
              >
                14.1.2022 16:00
              </p>
            </div>
            <div
              class="empity-slot css-lccpzd"
            >
              <div
                class="MuiBox-root css-1c6wt7s"
              >
                <div
                  class="MuiBox-root css-1agjfwn"
                >
                  <button
                    class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
                    tabindex="0"
                    type="button"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge css-tzssek-MuiSvgIcon-root"
                      data-testid="AddCircleIcon"
                      focusable="false"
                      style="color: blue;"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm5 11h-4v4h-2v-4H7v-2h4V7h2v4h4v2z"
                      />
                    </svg>
                    <span
                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1em9ut6-MuiGrid-root"
        >
          <div
            class="css-gg4vpm"
          >
            <button
              class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-sghohy-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="button"
            >
              Clear
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
            <button
              class="MuiLoadingButton-root MuiButton-root MuiButton-contained MuiButton-containedSuccess MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root css-13m85cd-MuiButtonBase-root-MuiButton-root-MuiLoadingButton-root"
              tabindex="0"
              type="submit"
            >
              Update
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
    </form>
  </div>,
]
`;
